x-pack/plugins/maps/public/routing/bootstrap/services/saved_gis_map.ts

```diff
@@ -18,73 +24,82 @@ import {
+// @ts-expect-error
 import { getStore } from '../../store_operations';
+import { MapStoreState } from '../../../reducers/store';
+import { LayerDescriptor } from '../../../../common/descriptor_types';
+
+export interface ISavedGisMap extends SavedObject {
+  layerListJSON?: string;
+  mapStateJSON?: string;
+  uiStateJSON?: string;
+  getLayerList(): LayerDescriptor[];
+  syncWithStore(): void;
+}
 
-export function createSavedGisMapClass(services) {
+export function createSavedGisMapClass(services: SavedObjectKibanaServices) {
   const SavedObjectClass = createSavedObjectClass(services);
 
-  class SavedGisMap extends SavedObjectClass {
-    static type = MAP_SAVED_OBJECT_TYPE;
+  class SavedGisMap extends SavedObjectClass implements ISavedGisMap {
+    public static type = MAP_SAVED_OBJECT_TYPE;
 
     // Mappings are used to place object properties into saved object _source
-    static mapping = {
+    public static mapping = {
       title: 'text',
       description: 'text',
       mapStateJSON: 'text',
       layerListJSON: 'text',
       uiStateJSON: 'text',
     };
-    static fieldOrder = ['title', 'description'];
-    static searchSource = false;
+    public static fieldOrder = ['title', 'description'];
+    public static searchSource = false;
 
-    constructor(id) {
+    public showInRecentlyAccessed = true;
+    public layerListJSON?: string;
+    public mapStateJSON?: string;
+    public uiStateJSON?: string;
+
+    constructor(id: string) {
       super({
         type: SavedGisMap.type,
         mapping: SavedGisMap.mapping,
         searchSource: SavedGisMap.searchSource,
         extractReferences,
-        injectReferences: (savedObject, references) => {
+        injectReferences: (savedObject: ISavedGisMap, references: SavedObjectReference[]) => {
           const { attributes } = injectReferences({
             attributes: { layerListJSON: savedObject.layerListJSON },
             references,
           });
 
           savedObject.layerListJSON = attributes.layerListJSON;
-
-          const indexPatternIds = references
-            .filter((reference) => {
-              return reference.type === 'index-pattern';
-            })
-            .map((reference) => {
-              return reference.id;
-            });
-          savedObject.indexPatternIds = _.uniq(indexPatternIds);
         },
 
         // if this is null/undefined then the SavedObject will be assigned the defaults
-        id: id,
+        id,
 
         // default values that will get assigned if the doc is new
         defaults: {
-          title: 'New Map',
+          title: i18n.translate('xpack.maps.newMapTitle', {
+            defaultMessage: 'New Map',
+          }),
           description: '',
         },
       });
-      this.showInRecentlyAccessed = true;
-    }
 
-    getFullPath() {
-      return getExistingMapPath(this.id);
+      this.getFullPath = () => {
```

Comment:
Why is this function getting created in the constructor?

Reply:
typescripting. Something about functional method versus prototypemethod.

Reply:
Sounds related to the fact that it's now an arrow function. Does it need to be? It doesn't look like this PR changes any calls to it

Reply:
for typescripting purposes, yes. I had to copy what [SavedDashboard](https://github.com/elastic/kibana/blob/master/src/plugins/dashboard/public/saved_dashboards/saved_dashboard.ts#L109) was doing to get TS to be happy.